test/acceptance/features/environment.py

```diff
@@ -49,9 +55,35 @@ def before_all(_context):
         _context.sbo_namespace = output
     else:
         assert False, f"TEST_ACCEPTANCE_START_SBO={start_sbo} is currently unsupported."
+    ctx.no_scenarios_failed = True
 
 
 def before_scenario(_context, _scenario):
     _context.bindings = dict()
     output, code = cmd.run(f'{ctx.cli} get ns default -o jsonpath="{{.metadata.name}}"')
     assert code == 0, f"Checking connection to OS cluster by getting the 'default' project failed: {output}"
+
+
+def after_scenario(_context, scenario):
+    if "namespace" in _context:
+        namespace = _context.namespace.name
+    elif os.getenv("TEST_NAMESPACE"):
+        namespace = os.getenv("TEST_NAMESPACE")
+    else:
+        print("No namespace set in context nor TEST_NAMESPACE env variable is set, skipping deletion")
+    if delete_namespace == "alwaysdelete":
+        delete_current_namespace(namespace)
+    elif (delete_namespace == "keepwhenfailed"):
+        if scenario.status == Status.passed and (ctx.no_scenarios_failed):
+            delete_current_namespace(namespace)
+        else:
+            ctx.no_scenarios_failed = False
+            print(f"Deleting namespace {namespace} skipped, since one of the scenarios failed.")
+    elif delete_namespace == "neverdelete":
+        print(f"Namespace {namespace} deletion skipped.")
+
+
+def delete_current_namespace(name):
+    output, code = cmd.run(f"{ctx.cli} delete namespace {name} --ignore-not-found")
```

Comment:
Please consider adding a timeout to the command so that the execution is not prolonged unnecesarily in case of failures to delete the namespace.